File: Day1/first.cc
#include <fstream>
#include <iostream>
#include <stdio.h>

int main(int argc, char **argv) {
    char cur, prev, first;
    int sum = 0;

    std::fstream fin("input.txt", std::fstream::in);

    fin >> first;
    prev = first;

    while (fin >> cur) {
        if (cur == prev) {
            sum += (int) cur - 48;
        }
        prev = cur;
    }

    if (cur == first) {
        sum += (int) cur - 48;
    }
    
    std::cout << "Sum: " << sum << "\n";
}

File: Day1/first.py
with open('input.txt') as f:
    line = list(f.readline().strip())
    nums = [int(x) for x in line]

    sum = 0

    for i in range(0, len(nums)):
        if (nums[i] == nums[(i + 1) % len(nums)]):
            sum += nums[i]

    print sum
